src/main/java/httpServer/http/response/HTTPResponseBuilder.java

```diff
@@ -0,0 +1,67 @@
+package httpServer.http.response;
+
+import httpServer.http.StatusCodes;
+import httpServer.http.request.HTTPRequest;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class HTTPResponseBuilder {
+
+    public String statusLine;
+    public Map<String, String> headers = new HashMap<>();
+    public String body;
+
+    public HTTPResponseBuilder setStatusLine(StatusCodes statusCode, HTTPRequest request) {
+        this.statusLine = request.httpVersionNumber + " " + statusCode.httpResponse;
+        return this;
+    }
+
+    public HTTPResponseBuilder setHeaderLine(String header, String value) {
+        return this;
+    }
+
+    public HTTPResponseBuilder setBody(String body) {
+        this.body = body;
+        return this;
+    }
+
+    public HTTPResponseBuilder makeAllowHeader(List methods) {
+        StringBuilder allowedMethods = new StringBuilder();
+        for (int i = 0; i < methods.size(); i++) {
```

Comment:
You can use `String.join` to just create the string from the `methods` array, without using a `for` loop.

Reply:
Ended up using a stream in a similar way to make the whole thing shorter!